docs/developer/release-cycle.md

```diff
@@ -0,0 +1,69 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`).
+
+**Note**: For the `v1.3` and `v1.4` release cycles we assume a ~ 4 months (~ 17 weeks) release cycle. This should be revisited after the `v1.4.0` release.
+
+The release cycle can be split up into the following phases:
+
+* Week 1-12: Active development
+    * Alpha releases based on the main branch can be released when necessary
```

Comment:
should we also mention monthly patch releases for the previous release?

Reply:
I have it in "After that"

I'm not sure if we should try to integrate montly patch releases of previous release cycles into the timline of the current release cycle. It's definitely something that should be mentioned in the tasks for the release team of the current release cycle though (but that's another doc/PR)

It's definitely something that has to be done. Currently it's just outside of the definition of a release cycle ("release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`)")

Reply:
Got it, I guess k8s doesn't include those in release timeline either

Reply:
Exactly

Reply:
I'll keep it here in this description only under "After that" to just have a description of one individual release cycle.

I'll try to do something with the timeline